stake-pool/program/src/processor.rs

```diff
@@ -1107,42 +1383,40 @@ mod tests {
         );
         let mut deposit_info = do_deposit(&mut pool_info, stake_balance, &mut pool_token_receiver);
 
-        let withdraw_amount = sol_to_lamports(5.0);
-        let tokens_to_burn: u64 = 5_000_000_000;
+        // Need to deposit more to cover deposit fee
+        let fee_amount = stake_balance * FEE_DEFAULT.numerator / FEE_DEFAULT.denominator;
+        let extra_deposit = (fee_amount * FEE_DEFAULT.denominator)
+            / (FEE_DEFAULT.denominator - FEE_DEFAULT.numerator);
+
+        let _extra_deposit_info =
+            do_deposit(&mut pool_info, extra_deposit, &mut pool_token_receiver);
 
         approve_token(
             &TOKEN_PROGRAM_ID,
             &pool_token_receiver.key,
             &mut pool_token_receiver.account,
             &pool_info.withdraw_authority_key,
             &pool_token_receiver.owner,
-            tokens_to_burn,
+            stake_balance,
         );
 
-        let stake_to_receive_key = Pubkey::new_unique();
-        let mut stake_to_receive_account =
-            Account::new(stake_balance, STAKE_ACCOUNT_LEN, &stake_program_id());
-
         let _result = do_process_instruction(
-            withdraw(
+            claim(
```

Comment:
nit: The rest of the tests using `do_withdraw` and `do_deposit` are really easy to read -- any chance you could make a `do_claim` function to refactor a lot of this logic?